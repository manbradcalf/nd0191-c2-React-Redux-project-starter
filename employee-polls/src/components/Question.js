import * as React from "react";
import Card from "@mui/material/Card";
import Button from "@mui/material/Button";
import Typography from "@mui/material/Typography";
import { connect } from "react-redux";
import { Avatar } from "@mui/material";
import { Box } from "@mui/material";
import { Link } from "react-router-dom";

const QuestionCard = ({ question, employees, authedUser, loading }) => {
  const votedAlready = employees[authedUser].answers[question.id]
    ? true
    : false;

  return (
    <Card
      sx={{
        backgroundColor: "#fff",
        borderRadius: 2,
        textAlign: "center",
        alignItems: "center",
        fontWeight: "light",
        m: 2,
      }}
    >

      <Box width="fit-content" sx={{ m: 1, p: 1 }}>
        <Avatar
          src={employees?.[question?.author]?.avatarURL}
          sx={{ margin: "auto" }}
        />
        <Typography
          sx={{
            fontStyle: "italic",
            m: "auto",
          }}
        >
          {question?.author}
        </Typography>
        <Typography
          sx={{
            fontStyle: "caption",
            fontSize: "x-small",
            m: "auto",
          }}
        >
          {new Date(question.timestamp).toLocaleDateString("en-us", {
            weekday: "long",
            year: "numeric",
            month: "short",
            day: "numeric",
          })}
        </Typography>
      </Box>
      <Typography sx={{ fontStyle: "bold" }} variant="h6">
        Would you rather...
      </Typography>

      <Typography sx={{ textAlign: "center", mb: 2 }}>
        {question?.optionOne.text}
      </Typography>
      <b>OR</b>
      <Typography sx={{ textAlign: "center", mt: 2 }}>
        {question?.optionTwo.text}
      </Typography>
      <div>
        <Link to={`/question/${question.id}`}>
          <Button
            value={question.id}
            variant={"contained"}
            sx={{ width: 1 / 4, m: 2 }}
          >
            {votedAlready ? "View" : "Vote"}
          </Button>
        </Link>
      </div>
    </Card>
  );
};

const mapStateToProps = ({ employees, questions, authedUser }) => ({
  employees,
  questions,
  authedUser,
});

export default connect(mapStateToProps)(QuestionCard);
