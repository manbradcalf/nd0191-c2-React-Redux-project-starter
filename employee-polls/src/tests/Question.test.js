import Question from '../components/Question';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { exampleState } from './exampleState';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
const mockStore = configureStore([]);
const store = mockStore(exampleState);

describe('Question', () => {
  it('renders successfully', () => {
    var component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Question question={exampleState.questions.xj352vofupe1dqz9emx13r} />
        </BrowserRouter>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
