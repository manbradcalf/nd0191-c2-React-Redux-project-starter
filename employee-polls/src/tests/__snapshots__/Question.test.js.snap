// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question renders successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-k7rbtc-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiBox-root css-a0bgn9"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular css-3cjoc6-MuiAvatar-root"
          >
            <img
              class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
              src="https://www.svgrepo.com/show/119054/avatar.svg"
            />
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1krkekw-MuiTypography-root"
          >
            mtsamis
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-guaezl-MuiTypography-root"
          >
            Sunday, Apr 30, 2017
          </p>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-5mf4hx-MuiTypography-root"
        >
          Would you rather...
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1tqac9q-MuiTypography-root"
        >
          deploy to production once every two weeks
        </p>
        <b>
          OR
        </b>
        <p
          class="MuiTypography-root MuiTypography-body1 css-100rmmz-MuiTypography-root"
        >
          deploy to production once every month
        </p>
        <div>
          <a
            href="/question/xj352vofupe1dqz9emx13r"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
              value="xj352vofupe1dqz9emx13r"
            >
              View
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-k7rbtc-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-a0bgn9"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular css-3cjoc6-MuiAvatar-root"
        >
          <img
            class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
            src="https://www.svgrepo.com/show/119054/avatar.svg"
          />
        </div>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1krkekw-MuiTypography-root"
        >
          mtsamis
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-guaezl-MuiTypography-root"
        >
          Sunday, Apr 30, 2017
        </p>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6 css-5mf4hx-MuiTypography-root"
      >
        Would you rather...
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1tqac9q-MuiTypography-root"
      >
        deploy to production once every two weeks
      </p>
      <b>
        OR
      </b>
      <p
        class="MuiTypography-root MuiTypography-body1 css-100rmmz-MuiTypography-root"
      >
        deploy to production once every month
      </p>
      <div>
        <a
          href="/question/xj352vofupe1dqz9emx13r"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-4pjbfc-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
            value="xj352vofupe1dqz9emx13r"
          >
            View
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
